src/Memstate.Core/Models/Graph/GraphModel.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Memstate.Models.Graph
+{
+    public partial class GraphModel
+    {
+        /// <summary>
+        /// Unique id generator, shared by nodes and edges
+        /// </summary>
+        private long _lastId;
+
+        //Nodes and edges
+        private readonly SortedDictionary<long, Node> _nodesById;
+        private readonly SortedDictionary<long, Edge> _edgesById;
+
+        [NonSerialized]
```

Comment:
We haven't added support for snapshots yet so NonSerialized could be removed for now.